### PR TITLE
Fix parameter validation in `test_full` and workflow dispatch

### LINKED ISSUE #289: Test full is currently failing due to parameter validation
### Description of the bug

Running the pipeline with `test _full`  results in:

```
* --input (https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/samplesheet_full.csv): Validation of file failed:
	-> Entry 1: Error for field 'id' (35147139): Sample name must be provided and cannot contain spaces
```

Based on this slack discussion (https://nfcore.slack.com/archives/CL88J906S/p1731922001766669) , a more lenient validation mode should fix the issue and let the samplesheet validation handle strings and numbers. 

### Command used and terminal output

```console
Workflow options
  v1_schema                 : true
Input/output options
  input                     : https://raw.githubusercontent.com/nf-core/test-datasets/demultiplex/samplesheet/1.3.0/samplesheet_full.csv
  outdir                    :./results/
Demultiplexing options
  demultiplexer             : bcl2fastq
Institutional config options
  config_profile_name       : Full test profile
  config_profile_description: Full test dataset to check pipeline function
Core Nextflow options
  revision                  : 1.5.3
  profile                   : test_full
```

### Relevant files

_No response_

### System information

_No response_

### PR DESCRIPTION
**Pull request by atrigila**
Closes #289 

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Make sure your code lints (`nf-core pipelines lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] `CHANGELOG.md` is updated.

### CODE REVIEW

**Comment by atrigila** [comment]
`lenientMode` caused new issues in the removal of adapters from Illumina samplesheet. I am going to address these.

**Comment by atrigila** [comment]
The error started to happen when we switched to nf-schema with the new template merge. It happens in the `test_full ` because it is the only profile having numeric IDs. In particular, when we use `lenientMode` the `ch_samplesheet` reads the meta as id: `java.lang.Integer(value: 35147139)`. However, when we try to join with the `ch_samplesheet_no_adapter`  (where we get the meta from the filename) the class is: `id: java.lang.String (value: 35147139)`.

File: nextflow.config [hunk 1]
```diff
@@ -193,12 +193,6 @@ profiles {
 
 }
 
-// Load base.config by default for all pipelines
-includeConfig 'conf/base.config'
-
-// Load modules.config for DSL2 module specific options
-includeConfig 'conf/modules.config'
```
Comment (maxulysse) [inline nextflow.config]: they should still be loaded in this file

File: conf/test_full.config [hunk 2]
```diff
@@ -19,3 +19,6 @@ params {
     demultiplexer = 'bcl2fastq'
     v1_schema     = true
 }
+
+// Load modules.config afterwards to load ext.args in samshee correctly
+includeConfig 'modules.config'
```
Comment (maxulysse) [inline conf/test_full.config:24]: No need to do that here

File: nextflow.config (hunk 1 above)
Comment (maxulysse) [reply nextflow.config]: you can include these 2 files later in this file.
Otherwise they will never be included, and we kinda need them

Comment (atrigila) [reply nextflow.config]: I did not delete these files. I tested moving them to a different location. I think this comment is from an old revision. This PR was still a draft in progress.

Comment (maxulysse) [reply nextflow.config]: Forgot I said anything then.
And sorry for commenting on a draft, just saw it in my emails

File: conf/test_full.config (hunk 2 above)
Comment (atrigila) [reply conf/test_full.config:24]: This is a work in progress. However, this commit was to demonstrate that this is the only way to make `test_full` and the other test work **with the current configuration**. 

I've observed that if `modules.config` is loaded before `test_full` (as in the current `nextflow.config`), it overrides the `params.v1_schema` defined in `test_full`, making the pipeline fail at the `samshee` step. 

If we do not add this here (and simply move `modules.config` below the test profile configuration in `nextflow.config`), the `test_full` works but all the other test profiles fail. 

So, this is one way to address it, but I was still going to explore other more elegant approaches, so any suggestions welcome :)

File: nextflow.config (hunk 1 above)
Comment (atrigila) [reply nextflow.config]: No worries and I appreciate the feedback :) I was doing a few tests and just did a few other immediate changes after this one.

**Comment by atrigila** [comment]
@FriederikeHanssen I made a minor change after your approval. I replaced the `modules.config` loading in the `test_full` to use only the samshee config, which I think improves clarity. If you’re okay with this change, we can go ahead and merge it :)

**Review by FriederikeHanssen — APPROVED** [review]

**Pull request merged**